lib/src/rules/unnecessary_lambdas.dart

```diff
@@ -90,14 +90,17 @@ class _Visitor extends SimpleAstVisitor<void> {
       final statement = body.block.statements.single;
       if (statement is ExpressionStatement &&
           statement.expression is InvocationExpression) {
-        _visitInvocationExpression(statement.expression, node);
+        _visitInvocationExpression(
+            statement.expression as InvocationExpression, node);
```

Comment:
We could reduce the number of casts by assigning `statement.expression` to a local before testing the type (thus allowing promotion to work).